pkg/proxy/iptables/proxier.go

```diff
@@ -652,9 +634,9 @@ func updateServiceMap(
 		changes.Lock()
 		defer changes.Unlock()
 		for _, change := range changes.items {
-			mergeSyncRequired, existingPorts := serviceMap.mergeService(change.current)
-			unmergeSyncRequired := serviceMap.unmergeService(change.previous, existingPorts, staleServices)
-			syncRequired = syncRequired || mergeSyncRequired || unmergeSyncRequired
+			existingPorts := serviceMap.merge(change.current)
```

Comment:
endpoints is simply unmerge old, then merge new.  why is this more complex?  I feel like I figured it out last week and then forgot.  Comment?